scripts/mbedtls_framework/build_tree.py

```diff
@@ -122,3 +129,13 @@ def guess_tf_psa_crypto_root(root: Optional[str] = None) -> str:
         return root
     else:
         raise Exception('TF-PSA-Crypto source tree not found')
+
+def is_mbedtls_3_6_branch(root: Optional[str] = None) -> bool:
+    """Whether root is the root directory of the Mbed TLS 3.6 branch """
+    if root is None:
+        root = os.getcwd()
+    if not looks_like_mbedtls_root(root):
+        return False
+    with open(os.path.join(root, 'include', 'mbedtls', 'build_info.h'), 'r') as f:
+        pattern = re.compile("#define MBEDTLS_VERSION_NUMBER.*0x0306")
```

Comment:
If method is meant to return true if and only if:

1. We are in an MbedTLS root
2. Because we are in an MbedTLS root the path 'include/mbedtls/build_info.h` exists
3. The `MBEDTLS_VERSION_NUMBER == 0x0306`

Wouldn't it be simpler to write like that?

~~~~~
def is_mbedtls36_branch():
  try:
     with open(os.path.join(root, 'include', 'mbedtls', 'build_info.h'), 'r') as f:
       if re.findall(r"#define MBEDTLS_VERSION_NUMBER.*0x0306", f.read()):
           return True
   except:
       pass
   return False

~~~~~

Reply:
I've tried to improve the function scope, its documentation and implementation. Please have another look, thanks.